### PR TITLE
fix(belongsTo): do not try to link when foreign key is null

### PR DESCRIPTION
**Pull request by nickescallon**
Fixes https://github.com/js-data/js-data/issues/369

### CODE REVIEW

**Comment by jmdobry** [comment]
Looks good, though it has merge conflicts that need to be resolved.

File: src/Relation/BelongsTo.js [hunk 1]
```diff
@@ -16,7 +16,8 @@ export const BelongsToRelation = Relation.extend({
       return
     }
     const relatedId = utils.get(record, this.foreignKey)
-    if (relatedId !== undefined) {
+
+    if (!utils.isUorN(relatedId)) {
```
Comment (jmdobry) [inline src/Relation/BelongsTo.js]: To simplify, I would just do `if (relatedId !== undefined && relatedId !== null) {`

File: src/utils.js [hunk 2]
```diff
@@ -1187,6 +1187,18 @@ http://www.js-data.io/v3.0/docs/errors#${code}`
   },
 
   /**
+   * Return whether the provided value is undefined or null.
```
Comment (jmdobry) [inline src/utils.js]: This new function isn't necessary, see my other comment.

**Comment by nickescallon** [comment]
sounds good, comments addressed.

**Pull request merged**